modules/social_features/social_post/src/Hooks/SocialPostFormHooks.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace Drupal\social_post\Hooks;
+
+use Drupal\Core\Entity\ContentEntityForm;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\hux\Attribute\Alter;
+use Drupal\social_post\Service\SocialPostHelperInterface;
+
+/**
+ * Replace hook: social_post_form_post_form_alter.
+ *
+ * @package Drupal\social_post\Hooks
+ */
+final class SocialPostFormHooks {
+
+  /**
+   * The social post helper service.
+   *
+   * @var \Drupal\social_post\Service\SocialPostHelperInterface
+   */
+  private SocialPostHelperInterface $socialPostHelper;
+
+  /**
+   * The account proxy.
+   *
+   * @var \Drupal\Core\Session\AccountProxyInterface
+   */
+  private AccountProxyInterface $currentUser;
+
+  /**
+   * Constructor.
+   *
+   * @param \Drupal\social_post\Service\SocialPostHelperInterface $social_post_helper
+   *   The social post helper.
+   * @param \Drupal\Core\Session\AccountProxyInterface $current_user
+   *   The account proxy.
+   */
+  public function __construct(
+    SocialPostHelperInterface $social_post_helper,
+    AccountProxyInterface $current_user,
+  ) {
+    $this->socialPostHelper = $social_post_helper;
+    $this->currentUser = $current_user;
+  }
+
+  /**
+   * From alter hook: replacement of social_post_form_post_form_alter.
+   *
+   * The method definition contains all standard parameters defined by the
+   * definition of the hook.
+   *
+   * @param array $form
+   *   The drupal form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   */
+  #[Alter('form_post_form')]
+  public function formPostFormAlter(array &$form, FormStateInterface $form_state): void {
+    $content = $this->socialPostHelper->buildCurrentUserImage();
+    $form_object = $form_state->getFormObject();
+
+    if ($form_object instanceof ContentEntityForm === FALSE) {
+      return;
+    }
```

Comment:
I think if `$form_object` is not a `ContentEntityForm` then we have an implementation error right? So we probably want to make this an `assert` if we expect to catch this in testing or throw an actual exception if we think this might happen at run-time but shows an implementation error. The alternative would likely be a subtle and hard to debug bug (the post form not being altered) due to something that's wrong in an unrelated system (the post form no longer being a content entity form).

This if-statement also causes `$content` to be calculated but never used, so that statement should likely be after this guard clause.